core/src/stages/compute_ik.cpp

```diff
@@ -231,10 +230,10 @@ void ComputeIK::compute() {
 	properties().performInitFrom(INTERFACE, s.start()->properties());
 	const auto& props = properties();
 
-	planning_scene::PlanningScenePtr sandbox_scene = s.start()->scene()->diff();
+	const planning_scene::PlanningSceneConstPtr& scene{ s.start()->scene() };
```

Comment:
Can you explain why it wasn't sufficient to get a new scene diff?

Reply:
The main culprit of the bug was that `isTargetPoseColliding` [changes the ACM](https://github.com/ros-planning/moveit_task_constructor/pull/286/files#diff-eb4791e6fc52e29c2b9fb96a606928a1bf587989c5ebf29af186a50137ecc159L117) but the scene was reused later on.
The rest of the patch changes the logic a bit to use what is actually needed.

Reply:
Now, I see. The `sandbox_scene` was reused for later collision checking:
https://github.com/ros-planning/moveit_task_constructor/blob/84ccd679d980e30b5365fa542f1fd0a72598b79b/core/src/stages/compute_ik.cpp#L368